lib/net.js

```diff
@@ -1485,7 +1497,9 @@ Server.prototype.listen = function() {
     return this;
   }
 
-  throw new Error('Invalid listen argument: ' + util.inspect(options));
+  throw new errors.Error('ERR_INVALID_OPT_VALUE',
```

Comment:
EDIT: `NaN` does not get this far, `null` does.

Reply:
We kind of stretched the `ERR_INVALID_OPT_VALUE` error here to fit in this specific case. Trying to minimize too specific errors codes that may not be used elsewhere. There is also a discussion about it above: https://github.com/nodejs/node/pull/14782#discussion_r134096383. If the current implementation is not sufficient we could go with a new error.

Reply:
@pmatzavin I think a new error would not be too specific because neither `ERR_INVALID_OPT_VALUE` nor `ERR_INVALID_ARG_TYPE` properly describe errors thrown by this kind of "overloading arguments & options" methods, and the messages that they produce are a bit confusing for these errors. Also there are at least more than one method that does this (`server.listen()` and `socket.connect()`).

Reply:
In `net.js` the `ERR_INVALID_OPT_VALUE` err constructor is directly called only (once) in `Server.prototype.listen`. IMHO the  first part of the msg `The value "{ port: null }" is invalid` accurately describes the cause of the error (it does not mention if the user provided that value or not). The second part of the message `for option "options"` may be confusing. Either way the error is  that a value is invalid inside the options object. I agree that a new error like `ERR_INVALID_OPTION_OBJECT` may be more accurate but i am not sure if it is necessary. Any more feedback-should we change this?

Reply:
@joyeecheung after looking at this further I somewhat agree with @joyeecheung that it is not very intuitive as it is right now. Even though the original error was not really any better. But our current errors mainly reflect the code they are executed in. So most messages are e.g. a clear 1-to-1 of the argument name that is used in the code. The user will have to check that line of code from time to time to understand what the explicit error stands for and this might be such a case.

I am not certain that finding a good solution for this should hold up the PR and I personally think we should try to rework the errors in general as soon as they are all ported over.